src/Lecture_4/Homework_4/Task_11.java

```diff
@@ -0,0 +1,25 @@
+package Lecture_4.Homework_4;
+
+import java.util.Scanner;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class Task_11 {
+
+    public static void main(String[] args) {
+
+        Scanner sn = new Scanner(System.in);
+
+        System.out.println("Please enter an email");
+        String input = sn.nextLine();
+
+        Pattern pattern = Pattern.compile("^[A-Za-z0-9+_.-]+@(.+)$");
+        Matcher matcher = pattern.matcher(input);
+
+        if (matcher.matches()) {
```

Comment:
try using:
^(?=.{1,64}@)[A-Za-z0-9_-]+(\.[A-Za-z0-9_-]+)*@[^-][A-Za-z0-9-]+(\.[A-Za-z0-9-]+)*(\.[A-Za-z]{2,})$

Reply:
It works much better. I will spend more time looking at regex topic.